src/event/radio.ts

```diff
@@ -14,19 +14,13 @@ export function walkRadio(
         ? `input[type="radio"][name="${window.CSS.escape(el.name)}"]`
         : `input[type="radio"][name=""], input[type="radio"]:not([name])`,
     ),
-  )
-  for (let i = group.findIndex(e => e === el) + direction; ; i += direction) {
-    if (!group[i]) {
-      i = direction > 0 ? 0 : group.length - 1
-    }
-    if (group[i] === el) {
-      return
-    }
-    if (isDisabled(group[i])) {
-      continue
-    }
-
-    focusElement(group[i])
-    instance.dispatchUIEvent(group[i], 'click')
+  ).filter(elt => !isDisabled(elt))
+  // 1 or 0 radio buttons found
+  if (group.length <= 1) {
+    return;
   }
+  // Multiple buttons found
+  const nextRadioElement = group[(group.indexOf(el) + direction + group.length) % group.length];
+  focusElement(nextRadioElement)
+  instance.dispatchUIEvent(nextRadioElement, 'click')
```

Comment:
```suggestion
  )
  for (let i = group.findIndex(e => e === el) + direction; ; i += direction) {
    if (!group[i]) {
      i = direction > 0 ? 0 : group.length - 1
    }
    if (group[i] === el) {
      return
    }
    if (isDisabled(group[i])) {
      continue
    }

    focusElement(group[i])
    instance.dispatchUIEvent(group[i], 'click')
    return
  }
```

Just adding the missing `return` prevents calling `isDisabled` on irrelevant elements.